src/main/java/org/cbio/gdcpipeline/tasklet/SetUpPipelineTasklet.java

```diff
@@ -98,4 +88,80 @@ private void createOutputDirectory() throws FileNotFoundException {
             }
         }
     }
+
+    private File createTempDirectory() throws Exception {
+        String tmp_path = System.getProperty(SYSTEM_TMP_DIR_PROPERTY);
+        File tmp_dir = new File(tmp_path, TMP_DIR);
+        if (tmp_dir.exists()) {
+            if (LOG.isErrorEnabled()) {
+                LOG.error("Temp directory already exists. Attempting to delete directory and its contents :" + tmp_dir.getAbsolutePath());
+            }
+            deleteTempDir(tmp_dir);
+        }
+        tmp_dir.mkdir();
+        return tmp_dir;
+    }
+
+    private void deleteTempDir(File tmp_dir) throws Exception {
+        File[] entries = tmp_dir.listFiles();
+        if (entries != null) {
+            for (File entry : entries) {
+                deleteTempDir(entry);
+            }
+        }
+        tmp_dir.delete();
+    }
+
+    private void extractCompressedFiles() throws Exception {
+        if (LOG.isInfoEnabled()) {
+            LOG.info("######### Extracting Compressed Files #########");
+        }
+        File temp_dir = createTempDirectory();
+        List<File> filesToExtract = getFilesToExtract();
+        if (!filesToExtract.isEmpty()) {
+            for (File extractFile : filesToExtract) {
+                File tmp_file;
+                try {
+                    tmp_file = File.createTempFile(extractFile.getName(), ".tmp", temp_dir);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    if (LOG.isErrorEnabled()) {
+                        LOG.error("Error creating temp file in : " + temp_dir.getAbsolutePath() + "\nSkipping File");
+                    }
+                    //TODO Skip ? Terminate ?
+                    continue;
+                }
+                try {
+                    FileInputStream fis = new FileInputStream(extractFile);
+                    FileOutputStream fos = new FileOutputStream(tmp_file);
+                    GZIPInputStream gzip = new GZIPInputStream(fis);
+                    byte[] buffer = new byte[1024];
+                    while (gzip.read(buffer) != -1) {
+                        fos.write(buffer);
+                    }
+                    fos.close();
+                    fis.close();
+                    gzip.close();
+                    String destination = sourceDir + File.separator + extractFile.getName().replace(".gz", "");
+                    Files.move(Paths.get(tmp_file.getAbsolutePath()), Paths.get(destination), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Yes there is no Hits object. Maybe during a re-run of the batch, we already might have an uncompressed file in the source directory. I was thinking to keep the decompression as default. It will just replace what's existing